example/poisson/poisson_driver.cpp

```diff
@@ -59,13 +59,47 @@ TaskCollection PoissonDriver::MakeTaskCollection(BlockList_t &blocks) {
   const int num_partitions = pmesh->DefaultNumPartitions();
   TaskRegion &solver_region = tc.AddRegion(num_partitions);
 
+  // make some storage for doing an Allreduce
+  reductions.resize(1);
+  // initialize it to zero
+  reductions[0].val = 0.0;
+
   for (int i = 0; i < num_partitions; i++) {
     // make/get a mesh_data container for the state
     auto &md = pmesh->mesh_data.GetOrAdd("base", i);
     auto &mdelta = pmesh->mesh_data.GetOrAdd("delta", i);
 
     TaskList &tl = solver_region[i];
 
+    // pass a pointer to the variable being reduced into
+    auto loc_red = tl.AddTask(none, poisson_package::SumMass<MeshData<Real>>, md.get(),
+                              &reductions[0].val);
+    // make it a regional dependency so dependent tasks can't execute until all lists do
+    // this
+    solver_region.AddRegionalDependencies(0, i, loc_red);
```

Comment:
why pass the region index instead of the `tl` reference?

Reply:
also why does the local reduce need the region dependency? Shouldn't only the test need it?

Reply:
re index for `tl` reference: it was just a choice.  this actually went in with my last PR.  I actually hate both the first two arguments, but prioritized getting the capability in over cleaning that interface piece up.

re why local reduce...: I think you're misinterpreting what the `AddRegionalDependencies` call is doing.  When you mark a `TaskID` like that, you're saying that dependent tasks (who in this example use `loc_rad` as a dependency) need to wait until all lists that registered that task need to finish before they can execute.  Example:

```c++
auto first = tl.AddTask(none, do_first);
region.AddRegionalDependencies(0, i, first);
auto second = tl.AddTask(first, do_second);
```

This code says that `do_second` won't be called until all the lists that mark `first` as a regional dependency have completed there calls to `do_first`.  The `do_first` call is just a normal task -- it doesn't know or care about the regional dependency thing.  I know it's a little confusing, but this is the basic info that needs to be provided to get this all working.  We can probably change some names to clarify things a bit.

So, back to the code in the Poisson driver.  What the code says is don't do the MPI_Allreduce until I'm done accumulating the result locally.  Does that make sense?

Reply:
> re index for `tl` reference: it was just a choice.  this actually went in with my last PR.  I actually hate both the first two arguments, but prioritized getting the capability in over cleaning that interface piece up.

Yeah fair enough. I agree it can stay for now. Its not a huge deal. But we'll want to clean it up Bernie downstream Codes start using it or it'll be too late.

> re why local reduce...:

Ah I see where my confusion lay! Somehow I had it stuck in my head that there was one task region per rank, but of course that's wrong. There's one per mesh block.